app/lib/nomis_client/base.rb

```diff
@@ -29,6 +29,9 @@ def token_request(method, path, params)
       rescue Faraday::ConnectionFailed, Faraday::TimeoutError => e
         Rails.logger.warn "Nomis Connection Error: #{e.message}"
         raise e
+      rescue OAuth2::Error => e
```

Comment:
This seems similar to the above rescue block, maybe we can combine them?

Reply:
The message is specific to the errors, I can't think of a neat way to combine them and retain the message granularity, but keen to learn!

Reply:
The only thing I can think of would be something like

```ruby
Rails.logger.warn "Nomis #{e.is_a? OAuth2::Error ? 'Oauth Client' : 'Connection'} Error: #{e.message}"
```

Reply:
That's fair, I was hoping the error message would give us the details of the error itself, but happy to keep as is